src/Raven.Studio/typescript/models/database/tasks/ongoingTaskQueueEtlEditModel.ts

```diff
@@ -0,0 +1,208 @@
+            let anyDirty = false;
+            this.transformationScripts().forEach(script => {
+                if (script.dirtyFlag().isDirty()) {
+                    anyDirty = true;
+                    // don't break here - we want to track all dependencies
+                }
+            });
+            return anyDirty;
+        });
+
+        this.dirtyFlag = new ko.DirtyFlag([
+                innerDirtyFlag,
+                this.taskName,
+                this.taskState,
+                this.mentorNode,
+                this.manualChooseMentor,
+                this.connectionStringName,
+                this.allowEtlOnNonEncryptedChannel,
+                this.optionsPerQueue,
+                scriptsCount,
+                hasAnyDirtyTransformationScript
+            ],
+            false, jsonUtil.newLineNormalizingHashFunction);
+    }
+    
+    private initValidation() {
+        this.initializeMentorValidation();
+
+        this.connectionStringName.extend({
+            required: true
+        });
+        
+        this.transformationScripts.extend({
+            validation: [
+                {
+                    validator: () => this.transformationScripts().length > 0,
+                    message: "Transformation Script is Not defined"
+                }
+            ]
+        });
+
+        this.optionsPerQueue.extend({
+            validation: [
+                {
+                    validator: () => {
+                        const names = this.optionsPerQueue().map(x => x.queueName()).filter(x => !!x);
+                        return new Set(names).size === this.optionsPerQueue().filter(x => !!x.queueName()).length;
+                    },
+                    message: "Duplicate queue names"
+                }
+            ]
+        }); 
+
+        this.validationGroup = ko.validatedObservable({
+            connectionStringName: this.connectionStringName,
+            mentorNode: this.mentorNode,
+            transformationScripts: this.transformationScripts,
+            optionsPerQueue: this.optionsPerQueue
+        });
+    }
+
+    update(dto: Raven.Client.Documents.Operations.OngoingTasks.OngoingTaskQueueEtlDetails) {
+        super.update(dto);
+
+        if (dto.Configuration) {
+            this.connectionStringName(dto.Configuration.ConnectionStringName);
+            this.transformationScripts(dto.Configuration.Transforms.map(x => new ongoingTaskQueueEtlTransformationModel(x, false, false)));
+            this.manualChooseMentor(!!dto.Configuration.MentorNode);
+            
+            if (dto.Configuration.Queues) {
+                dto.Configuration.Queues.forEach(x => {
+                    const queueOptions = new queueOptionsModel(x.Name, x.DeleteProcessedDocuments);
+                    this.optionsPerQueue.push(queueOptions);
+                });
+            }
+        }
+    }
+
+    toDto(broker: QueueBroker): Raven.Client.Documents.Operations.ETL.Queue.QueueEtlConfiguration { 
```

Comment:
I think broker should be field of this class. Not external dependency.

Reply:
oh I can see we have 2 subclasses - so let's make this method protected if there will be name clash then this method can be:
protected toDtoInternal(broker: QueueBroker) and then you can have public methods toDto() in child classes.